lib/src/command/remove.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:yaml_edit/yaml_edit.dart';
+
+import '../command.dart';
+import '../exceptions.dart';
+import '../io.dart';
+
+/// Handles the `remove` pub command.
+class RemoveCommand extends PubCommand {
+  @override
+  String get name => 'remove';
+  @override
+  String get description => 'Removes a dependency from the current package.';
+  @override
+  String get invocation => 'pub remove <package>';
+  @override
+  String get docUrl => 'https://dart.dev/tools/pub/cmd/pub-remove';
+
+  RemoveCommand();
+
+  @override
+  Future run() async {
+    if (argResults.rest.isEmpty) {
+      usageException('Must specify a package to be removed.');
+    }
+
+    /// Update the pubspec.
+    _removePackagesFromPubspec(argResults.rest);
```

Comment:
Perhaps use `Set(argResults.rest).toList()` to deduplicate.

Reply:
That might be nice, but it if we're nit picking I would suggest also removing in the order given on the CLI. That makes errors easier to understand.

Reply:
Just to confirm that I am understanding this correctly, does `Set.from(argResults.rest).toList()` not preserve the order given on the CLI?

Reply:
Yes, to my knowledge it preserves order but removes duplicates.

I am no longer sure if we want to remove duplicates - I guess they could indicate an error, and we should really report that?